php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -2835,8 +2889,32 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         if ($request->getMethod() === "POST" && !isset($req['ClearInstrument'])) {
             $this->save();
         }
-        return (new \Laminas\Diactoros\Response())
+
+        // Disable form if data entry is complete
+        $user      = $request->getAttribute("user");
+        $sessionID = $request->getQueryParams()["sessionID"];
+        $timepoint = \Timepoint::singleton($sessionID);
+
+        // create an instrument status object
+        $status = new \NDB_BVL_InstrumentStatus;
+        $status->select($this->commentID);
+
+        // freeze the form to prevent data entry
+        if ($status->getDataEntryStatus() != 'In Progress'
+            || !$user->hasPermission('data_entry')
+            || (isset($timepoint)
+            && !in_array(
+                $timepoint->getData("CenterID"),
+                $user->getData("CenterIDs")
+            ))
+        ) {
+
+            if ($this->preview !== true && $this->DataEntryType !== 'DirectEntry') {
+                $this->freeze();
+            }
+        }
```

Comment:
Why do we have nested IFs ? there is no alternate to the second if, it could just be combined with the first

Reply:
just realized the `&&` on 2912 should likely be a `||` operator